src/flownet/ahm/_assisted_history_matching.py

```diff
@@ -1,18 +1,25 @@
-import os
-import json
-import re
+import argparse
+import concurrent.futures
 import glob
-import subprocess
+import itertools
+import json
+import math
+import os
 import pathlib
-import argparse
+import re
 import shutil
-import concurrent.futures
+import subprocess
 from typing import List, Dict, Optional, Tuple
+import yaml
+import matplotlib.pyplot as plt
+from matplotlib.ticker import ScalarFormatter
+from sklearn.metrics import mean_squared_error, mean_absolute_error, r2_score
```

Comment:
When we first change the order here of imports, maybe we can group them according to PEP8. 
> Imports should be grouped in the following order:
> 
> Standard library imports.
> Related third party imports.
> Local application/library specific imports.
> You should put a blank line between each group of imports.